spec/models/institutions_archive_spec.rb

```diff
@@ -0,0 +1,144 @@
+
+        # version 4
+        create_production_version
+        create :institution, version: current_production_number
+
+        initial_institution_count = 4
+        institution_count = 1
+        institutions_archive_count = 3
+
+        archive_test(initial_institution_count,
+                     institution_count,
+                     institutions_archive_count)
+      end
+
+      it 'archives multiple old versions with preview versions greater than current production', js: true do
+        # version 3
+        create_production_version
+        create :institution, version: current_production_number
+
+        # version 4
+        create_production_version
+        create :institution, version: current_production_number
+
+        # preview version 5
+        create :version, :preview
+        create :institution, version: current_preview_number
+
+        initial_institution_count = 5
+        institution_count = 2
+        institutions_archive_count = 3
+
+        archive_test(initial_institution_count,
+                     institution_count,
+                     institutions_archive_count)
+      end
+    end
+
+    context 'when not successful' do
+      it 'returns an error message' do
+        error_message = 'BOOM!'
+        allow(InstitutionsArchive).to receive(:create_archives).and_raise(StandardError, error_message)
+        expect(Rails.logger).to receive(:error).with("There was an error of unexpected origin: #{error_message}")
+        InstitutionsArchive.archive(Version.current_production)
+      end
+
+      it 'logs errors at the database level' do
+        error_message = 'BOOM!'
+        pg_result = double('PG::Result Double', error_message: error_message)
+        pg_error = double('PG::Error Double', result: pg_result)
+
+        statement_invalid = ActiveRecord::StatementInvalid.new('message', pg_error)
+        statement_invalid.set_backtrace(%(backtrace))
+
+        allow(InstitutionsArchive).to receive(:create_archives).and_raise(statement_invalid)
+        expect(Rails.logger).to receive(:error)
+          .with("There was an error occurring at the database level: #{error_message}")
+        InstitutionsArchive.archive(Version.current_production)
+      end
+    end
+  end
+
+  # private methods
+  private
+
+  def archive_test(initial_institution_count,
+                   institution_count,
+                   institutions_archive_count)
+    expect(Institution.count).to eq(initial_institution_count)
+    expect(InstitutionsArchive.count).to eq(0)
+
+    InstitutionsArchive.archive(Version.current_production)
+
+    expect(Institution.count).to eq(institution_count)
+    expect(Institution.where('version >= ?', current_production_number).size).to eq(institution_count)
+
+    expect(InstitutionsArchive.count).to eq(institutions_archive_count)
+    expect(InstitutionsArchive.where('version < ?', current_production_number).size).to eq(institutions_archive_count)
+  end
+
+  def create_production_version
```

Comment:
This can be done in a `before do ... end` block

Reply:
At line 14 there is `before do ... end` block that calls this function twice, this is also used to create additional versions in individual tests for creating additional production versions